### PR TITLE
[AMD] Fix FP8 assertion failure in aiter MLA decode by falling back to self.k_scale

### PR DESCRIPTION
**Pull request by yichiche**
## Motivation

When `layer.k_scale` is `None` (the default in `RadixAttention`), the aiter ASM MLA kernel asserts `q_scale.has_value() && kv_scale.has_value()` for FP8 Q tensors. Fall back to `self.k_scale` (initialized to `tensor([1.0])`) at all 4 `mla_decode_fwd` call sites, matching the pattern used by `flashmla_backend`.
This issue was found in Kimi-K2.5 but not in DeepSeek-R1. It must be resolved to enable the FP8 KV cache for Kimi-K2.5 / Kimi-K2.5 mxfp4 on MI355.
## Modifications

- Fix assertion failure `q_scale.has_value() && kv_scale.has_value()` in `aiter/csrc/py_itfs_cu/asm_mla.cu:206` when running MLA models (e.g., Kimi-K2.5, DeepSeek-R1) with `--attention-backend aiter --kv-cache-dtype fp8_e4m3`
- Root cause: The aiter ASM MLA decode kernel requires valid `q_scale` and `kv_scale` tensors when Q is FP8, but all 4 `mla_decode_fwd` call sites in `aiter_backend.py` pass `layer.k_scale` directly, which defaults to `None` in `RadixAttention`. The `None` propagates as an empty `std::optional` to the C++ kernel, triggering the assertion during CUDA graph capture.
- Fix: Fall back to `self.k_scale` (initialized to `torch.tensor([1.0])` at `AiterAttnBackend.__init__`) when `layer.k_scale is None`, at all 4 call sites:
  - `forward_extend` → target\_verify path (line 1650)
  - `forward_extend` → draft\_extend non-graph path (line 1701)
  - `forward_extend` → draft\_extend graph path (line 1731)
  - `forward_decode` (line 1867)
- This matches the fallback pattern already used by `flashmla_backend.py`

## Accuracy Tests

```
python3 benchmark/gsm8k/bench_sglang.py --num-questions 2000 --parallel 1000 --num-shots 5 --port 9000
```

| Metric | Value |
|---|---|
| Accuracy | 0.934 |
| Invalid | 0.001 |
| Latency | 1105.744 s |
| Output throughput | 120.129 token/s |

## Benchmarking and Profiling

**Server:** MI355 x 8
**IL:** 8k
**OL:** 1k
**Concurrencies:** 1

```bash
MODEL=./Kimi-K2.5
SGLANG_AITER_MLA_PERSIST=1 python3 -m sglang.launch_server \
  --model-path "${MODEL}" \
  --tensor-parallel-size 4 \
  --trust-remote-code \
  --chunked-prefill-size 131072 \
  --host 0.0.0.0 \
  --port 9000 \
  --log-requests \
  --disable-radix-cache \
  --mem-fraction-static 0.8 \
  --max-running-requests 64 \
  --kv-cache-dtype fp8_e4m3 \
  --attention-backend aiter 2>&1 | tee tmp.log
```

```
============ Serving Benchmark Result ============
Backend:                                 sglang    
Traffic request rate:                    inf       
Max request concurrency:                 1         
Successful requests:                     8         
Benchmark duration (s):                  305.89    
Total input tokens:                      64768     
Total input text tokens:                 64768     
Total generated tokens:                  8192      
Total generated tokens (retokenized):    8192      
Request throughput (req/s):              0.03      
Input token throughput (tok/s):          211.73    
Output token throughput (tok/s):         26.78     
Peak output token throughput (tok/s):    29.00     
Peak concurrent requests:                2         
Total token throughput (tok/s):          238.51    
Concurrency:                             1.00      
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   38231.62  
Median E2E Latency (ms):                 38230.12  
P90 E2E Latency (ms):                    38240.68  
P99 E2E Latency (ms):                    38241.71  
---------------Time to First Token----------------
Mean TTFT (ms):                          1799.84   
Median TTFT (ms):                        1799.62   
P99 TTFT (ms):                           1808.76   
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          35.61     
Median TPOT (ms):                        35.61     
P99 TPOT (ms):                           35.62     
---------------Inter-Token Latency----------------
Mean ITL (ms):                           35.61     
Median ITL (ms):                         35.61     
P95 ITL (ms):                            35.68     
P99 ITL (ms):                            35.76     
Max ITL (ms):                            36.79     
==================================================
```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Review by kkHuang-amd — COMMENTED** [review]
What is the difference between test/registered/amd/accuracy/mi35x/test_kimi_k25_mxfp4_eval_mi35x.py and test_kimi_k25_mxfp4.py.

Look like the configuration is almost similar in accuracy testing

**Comment by yichiche** [comment]
> What is the difference between test/registered/amd/accuracy/mi35x/test_kimi_k25_mxfp4_eval_mi35x.py and test_kimi_k25_mxfp4.py.
> 
> Look like the configuration is almost similar in accuracy testing

We have PR testing and nightly testing. According to YC's feedback, tests taking longer than 15 minutes will move from PR testing (every PR) to nightly testing (after merging to sglang main). Since the mxfp4 version of Kimi2.5 is small enoguh, it will allow us to add PR testing (nightly testing is by default).

**Review by HaiShaw — APPROVED** [review]
